apis/stacks/v1alpha1/types.go

```diff
@@ -244,6 +270,29 @@ type StackSpec struct {
 	Permissions     PermissionsSpec `json:"permissions,omitempty"`
 }
 
+// ServiceAccountLabels guarantees a map of label strings
+// from a StackSpec
+func (spec StackSpec) ServiceAccountLabels() map[string]string {
+	annotations := map[string]string{}
+	if sa := spec.Controller.ServiceAccount; sa != nil {
+		for k, v := range sa.Annotations {
+			annotations[k] = v
+		}
+	}
+	return annotations
+}
+
+// ImagePullSecrets guarantees a slice of LocalObjectReference from the
+// StackSpec's ServiceAccount.ImagePullSecrets ready for use in Deployment and
+// ServiceAccount objects
+func (spec StackSpec) ImagePullSecrets() []corev1.LocalObjectReference {
```

Comment:
I could avoid defining `ImagePullSecrets` for both StackSpec and StackInstallSpec if the method acts on `ServiceAccountOptions`.  Doing so might invigorate the cyclomatic complexity check that I was trying to avoid by making this method available on the higher level Spec.  The `ServiceAccount` properties on these specs are nil'able, and the `if != nil` check increases complexity in the functions that rely on this.